### PR TITLE
WIP Allow specifying glob pattern to ignore config option

### PR DESCRIPTION
**Pull request by topecongiro**
cc #3243, #3395.

### CODE REVIEW

File: src/ignore.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+use globset::{self, Glob, GlobSet, GlobSetBuilder};
+
+use crate::config::{FileName, IgnoreList};
+
+pub struct IgnorePathSet {
```
Comment (scampi) [inline src/ignore.rs]: Can you write few words documenting this struct ?

**Comment by topecongiro** [comment]
[ ] We should use [ignore](https://docs.rs/ignore/0.4.6/ignore/) crate instead

**Comment by rchaser53** [comment]
I don't have the permission to push to ```topecongiro:ignore-with-glob```. So I create new branch from ```topecongiro:ignore-with-glob``` and will send the PR.

**Comment by topecongiro** [comment]
Closing in favor of #3488.

**Pull request closed without merging**